LanguageFeatures/Patterns/logical_and_A06_t01.dart

```diff
@@ -40,15 +40,15 @@ main() {
       break;
     default:
   }
-  Expect.equals("Square.area=1;Square.area=2;", shape2.log);
+  Expect.equals("Square.area=1;=2;", shape2.log);
```

Comment:
Do you know positively that it will behave like this? I still think it will not be possible to detect that the matching with 2 occurred:

`Square(area: 2)` is an object pattern containing a pattern field `area: 2` which contains a constant pattern `2`.

Matching with an object pattern will check that we have a `Square` (or a subtype of that), then call the getter `area` yielding an object `r` (a `Unit`, I think), and then it will evaluate `2 == r` (not `r == 2`).

I guess the rationale is that a constant pattern should rely on the constant to get the equality result, because "we can trust `==` of the constant"; in contrast, we could get funny behaviors out of the equality test if the implementation of operator `==` in `r` gets to decide.

So `2` gets to decide on the equality, and operator `==` of the `Unit` doesn't get executed, and hence nothing is written to the log for that whole `area: 2` pattern field matching step.

Reply:
Yes, matching algorithm described at https://github.com/dart-lang/language/blob/master/accepted/future-releases/0546-patterns/feature-specification.md#matching-refuting-and-destructuring ends with 2 == `r`, where `r` is `Unit`. I have to rewrite these tests. Thank you

Reply:
Thanks! It is indeed tricky. ;-)